Automattic/wp-calypso
# Store Orders: Add "Resend Invoice" functionality to pending-payment orders

See #20794 – This PR gets the state & data requests up for manually sending an invoice email via the new API endpoint in https://github.com/Automattic/wc-calypso-bridge/pull/21. This also adds a "Resend Invoice" button to the header of all pending-payment orders:

<img width="896" alt="resend-button" src="https://user-images.githubusercontent.com/541093/34631566-f2c2833e-f23e-11e7-8349-1d78d75d7260.png">

Clicking this triggers the API and should send off an invoice (once the updated wc-calypso-bridge is pushed, or if you've got master running on your site - otherwise it'll just error-notice at you)

There's a new action, reducer subtree, and selector for tracking the status of the invoice request; along with a new data layer handler to send the request.

**To test**

- Run the tests `npm run test-client client/extensions/woocommerce/state`
- Check out the button on an order -  `/store/orders/:site`, and pick an order pending payment
- You should see the Resend Invoice button in the top bar
- Clicking it will trigger an action, and if you have the updated bridge plugin, send an email invoice.
- The Activity Log will also be updated immediately with the note that you've sent the invoice. 

@kellychoffman @jameskoster Let me know if you meant a different location by "'Resend invoice' button appears in header." (also, I'll have a separate PR for adding the button to the success notice)
